gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -364,7 +430,7 @@ public void checkAccess(Path path, AccessMode... modes) throws IOException {
     if (cloudPath.seemsLikeADirectoryAndUsePseudoDirectories()) {
       return;
     }
-    if (gcsService.getMetadata(cloudPath.getGcsFilename()) == null) {
+    if (storage.get(cloudPath.getBlobId()) == null) {
```

Comment:
Nit, you can pass `Storage.BlobGetOption.fields()` to minimize the response size.